etc/checkstyle.xml

```diff
@@ -73,7 +71,7 @@ under the License.
       <module name="RightCurly"/>
       <module name="RightCurly">
          <property name="option" value="alone"/>
-         <property name="tokens" value="CLASS_DEF, METHOD_DEF, CTOR_DEF, LITERAL_FOR, LITERAL_WHILE, STATIC_INIT, INSTANCE_INIT"/>
+         <property name="tokens" value="CLASS_DEF, METHOD_DEF, CTOR_DEF, LITERAL_FOR, LITERAL_WHILE, LITERAL_DO, STATIC_INIT, INSTANCE_INIT"/>
```

Comment:
It seems we can add
      <module name="RightCurly">
         <property name="option" value="same"/>
         <property name="tokens" value="LITERAL_DO"/>
      </module>

to keep the old format of a do while, with the while on same line, and have it checked by check style , i assume this is what you'd like